### PR TITLE
ScaleEntity Module

### PR DESCRIPTION
**Pull request by druiz17**
This Module adds the ability to scale unlocked entities that are grab-able or dynamic

### CODE REVIEW

**Comment by druiz17** [comment]
Test plan
- create two grab-able objects, dynamic and non-dynamic
- grab the object with one hand with the bumper trigger, then grab the same object with the other hand using the bumper trigger aswell
- move your hands in or out...you should see the entity scale accordingly.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3418/HighFidelity-Beta-PR11401-e2921661e9343b1d090295a1598b453cd58fb4dd.dmg

**Comment by druiz17** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3423/HighFidelity-Beta-PR11401-e2921661e9343b1d090295a1598b453cd58fb4dd.dmg

**Comment by druiz17** [comment]
test this please

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3432/HighFidelity-Beta-PR11401-e2921661e9343b1d090295a1598b453cd58fb4dd.dmg

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3440/HighFidelity-Beta-PR11401-e2921661e9343b1d090295a1598b453cd58fb4dd.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3440/HighFidelity-Beta-PR11401-e2921661e9343b1d090295a1598b453cd58fb4dd.dmg

**Comment by Nex-Pro** [comment]
Testing this Win10 and Vive

**Comment by Nex-Pro** [comment]
Not sure if pass or fail as I noticed a difference in haptic not mentioned in the test notes.

Test plan

create two grab-able objects, dynamic and non-dynamic **CHECK**
grab the object with one hand with the grip(edited) triggers, then grab the same object with the other hand using the bumper trigger aswell**CHECK**
move your hands in or out...you should see the entity scale accordingly. **SUCCESS**

Works for both dynamic and dynamic.

Dynamic scales quietly.
***Non dynamic scales with a lot of haptic pulses on the second hand.***

**Comment by Nex-Pro** [comment]
This isn't so great once the entity becomes large. But seems to work fine for smaller ents.

**Comment by druiz17** [comment]
The lots of haptic feedback is a different issue not related to this PR

**Comment by Nex-Pro** [comment]
@druiz17 Cool

File: scripts/system/controllers/controllerModules/scaleEntity.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        this.otherModule = function() {
+            return this.hand === dispatcherUtils.RIGHT_HAND ? leftScaleEntity : rightScaleEntity;
+        };
+
+        this.bumperPressed = function(controllerData) {
+            if ( controllerData.secondaryValues[this.hand] > dispatcherUtils.BUMPER_ON_VALUE) {
+                return true;
+            }
+            return false;
+        };
```
Comment (AndrewMeadows) [inline scripts/system/controllers/controllerModules/scaleEntity.js]: Why not just:
```
        this.bumperPressed = function(controllerData) {
            return ( controllerData.secondaryValues[this.hand] > dispatcherUtils.BUMPER_ON_VALUE); 
        };
```

Comment (druiz17) [reply scripts/system/controllers/controllerModules/scaleEntity.js]: That makes more sense.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3515/HighFidelity-Beta-PR11401-65e0c99618bf05b704f97715b516cb53ceb43d9c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3515/HighFidelity-Beta-PR11401-65e0c99618bf05b704f97715b516cb53ceb43d9c.exe

**Comment by anshuman64** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3519/HighFidelity-Beta-PR11401-65e0c99618bf05b704f97715b516cb53ceb43d9c.exe

**Comment by druiz17** [comment]
test this please

**Comment by druiz17** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3547/HighFidelity-Beta-PR11401-65e0c99618bf05b704f97715b516cb53ceb43d9c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3547/HighFidelity-Beta-PR11401-65e0c99618bf05b704f97715b516cb53ceb43d9c.exe

**Pull request merged**